### PR TITLE
[WIP][5.0] Dark Mode support

### PR DESCRIPTION
**Pull request by wilsonge**
Please note this is still miles away from even being ready for testing - but is just here so that the work is being done transparently and is able to be tracked for maintainers & others.

### Summary of Changes
Enables dark mode support for Joomla 5 on the back of the Bootstrap 5.3 upgrade of J4.

### Testing Instructions
This requires compiling of CSS so won't work with regular patchtester

Use dark mode on your browser (usually exposed through an OS Setting) and browse around the backend validating that visually dark mode is enabled and consistent.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
thanks for starting this pr, I'm adding a screenshot for now 
image

**Comment by wilsonge** [comment]
> Please note this is still miles away from even being ready for testing

:P

**Comment by web-eau-net** [comment]
I have tested this item :red_circle: unsuccessfully on 4069ac1b5729b7335ee3a96cf14bf897a56eeb03<br /><br />Dark mode activated in Opera 101.0.4843.43 but does not affect the backend :( <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41409">issues.joomla.org/tracker/joomla-cms/41409</a>.</sub>

**Comment by wilsonge** [comment]
Thankyou that is actually very helpful!! I've fixed tabs in article edit and the quickicons.

I will work my way through the rest of your list. Thankyou. thankyou. thankyou!

**Comment by wilsonge** [comment]
So not saying I'm done by any means (but it's definitely slowly getting there).

I've reached the point where we need to think about how extensions interact with this without necessarily becoming hard bootstrap coupled. So as an example we can look at media manager integration (component), or extension drag and drop (plugin) and harder still editor theme changing (tinymce/codemirror - harder because we need to decide if a custom theme even supports dark mode - we know the default ones do - as @dgrammatiko showed in his PR)

I'd appreciate some thoughts on this from @dgrammatiko , @HLeithner, @laoneo or anyone else for that matter

**Comment by dgrammatiko** [comment]
> I've reached the point where we need to think about how extensions interact with this without necessarily becoming hard bootstrap coupled.

That's a great call for design tokens. A design system backing the tokens would also be nice

>  harder because we need to decide if a custom theme even supports dark mode

It's not that hard. The moving parts here are:
- DocumentElement attribute that denotes if a theme is forced (custom ones other than light/dark are always forced!)
- DocumentElement attribute that has the name of the theme (light/dark/dimitris) that would be picked from the CSS/JS
- Server side rendered parts just follow the design token (basically Joomla named CSS vars)
- Client side apps (tinyMCE/etc) just have to have the logic for forced/OS options (check the [Muta](https://github.com/dgrammatiko/muta) implementation)

**Comment by wilsonge** [comment]
I’m not using themes directly in this version. Just media queries. But that’s going to have to change for extension integrations with the template. Can just be an options setting I guess - but doubt CSP will like toggling css files based on the media query. So likely somehow declared in PHP??

File: build/media_source/templates/administrator/atum/scss/_variables.scss [hunk 1]
```diff
@@ -232,13 +233,18 @@ $input-btn-submenu-icon-distance:   1rem;
...
 $form-select-bg-size:              calc(max(100%, 116rem));
 $form-select-indicator:            url("../images/select-bg.svg");
 $form-select-indicator-rtl:        url("../images/select-bg-rtl.svg");
 $form-select-indicator-active:     url("../../../images/select-bg.svg");
 $form-select-indicator-active-rtl: url("../../../images/select-bg-rtl.svg");
+$form-select-indicator-dark:       url("../images/select-bg-dark.svg");
+$form-select-indicator-rtl-dark:   url("../images/select-bg-rtl-dark.svg");
 $form-select-background:           $form-select-indicator no-repeat right center / $form-select-bg-size; // Used so we can have multiple background elements (e.g., arrow and feedback icon)
 $form-select-background-rtl:       $form-select-indicator-rtl no-repeat left center / $form-select-bg-size; // Used so we can have multiple background elements (e.g., arrow and feedback icon)
+$form-select-background-dark:      $form-select-indicator-dark no-repeat right center / $form-select-bg-size; // Used so we can have multiple background elements (e.g., arrow and feedback icon)
```
Comment (brianteeman) [inline build/media_source/templates/administrator/atum/scss/_variables.scss]:
```suggestion
$form-select-background-dark:      $form-select-indicator-dark no-repeat right center / $form-select-bg-size; // Used so we can have multiple background elements (e.g. arrow and feedback icon)
```

File: build/media_source/templates/administrator/atum/scss/_variables.scss [hunk 2]
```diff
@@ -232,13 +233,18 @@ $input-btn-submenu-icon-distance:   1rem;
...
+$form-select-background-rtl-dark:  $form-select-indicator-rtl-dark no-repeat left center / $form-select-bg-size; // Used so we can have multiple background elements (e.g., arrow and feedback icon)
```
Comment (brianteeman) [inline build/media_source/templates/administrator/atum/scss/_variables.scss]:
```suggestion
$form-select-background-rtl-dark:  $form-select-indicator-rtl-dark no-repeat left center / $form-select-bg-size; // Used so we can have multiple background elements (e.g. arrow and feedback icon)
```

**Comment by rabidgrowth** [comment]
(I am writing to you with Google Translate)

I thank you for all the work you did @wilsonge !

Regarding the new expansions, you put a lot of thought into it. See what Bootstrap says here: https://getbootstrap.com/docs/5.3/customize/color-modes/#sass-mixins

In my templates with Bootstrap 5.3 I do something like the following:

```
@import 'sections/home';

@include color-mode(dark) {
@import 'sections/home_dark';
}
```

In _home.scss I have the custom CSS I want for lightmode. This has everything in it, flexbox, margins, padding, and natural colors. In _home_dark.scss I only have colors. Bootstrap's color-mode automatically plays with $color-mode-type. If I change my mind, I recompile my SCSS. Simple and quick, without much fuss.

Don't overcomplicate it. Most sites are made by small companies with 2-3 people. Unnecessary complexity kills our profit. The customer pays by the piece, not by the hour.

**Comment by HLeithner** [comment]
After CI fix I merge this for beta1, we know additional things are needed but for now it's a good start.

**Comment by wilsonge** [comment]
CI Fixed

**Comment by HLeithner** [comment]
Thanks, docs needed

File: build/media_source/templates/administrator/atum/scss/_variables-dark.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  focus-shadow:                    var(--gray-800),
+  media-manager-overlay-bg:        var(--template-bg-dark-90),
+  media-manager-infobar-dt-color:  rgba(255, 255, 255, .54),
+  media-manager-overlay-header-bg: var(--template-bg-dark-80),
+) !default;
+
+$link-hover-color-dark: lighten($light-blue, 20%);
+
+$form-select-bg-dark:              var(--template-bg-dark);
+$form-select-background-dark:      $form-select-indicator-dark no-repeat right center / $form-select-bg-size; // Used so we can have multiple background elements (e.g. arrow and feedback icon)
```
Comment (richard67) [inline build/media_source/templates/administrator/atum/scss/_variables-dark.scss:12]: This is a syntax error, you cannot divide `$form-select-indicator-dark no-repeat right center`.

File: build/media_source/templates/administrator/atum/scss/_variables-dark.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+$form-select-background-rtl-dark:  $form-select-indicator-rtl-dark no-repeat left center / $form-select-bg-size; // Used so we can have multiple background elements (e.g. arrow and feedback icon)
```
Comment (richard67) [inline build/media_source/templates/administrator/atum/scss/_variables-dark.scss:13]: This is a syntax error, you cannot divide `$form-select-indicator-rtl-dark no-repeat left center`.

**Comment by brianteeman** [comment]
less haste more speed

**Comment by Quy** [comment]
41409-config
41409-dashboard
41409-inline-help

**Comment by Quy** [comment]
41409-media

**Pull request merged**